perun-core/src/main/java/cz/metacentrum/perun/core/implApi/modules/attributes/ResourceGroupAttributesModuleAbstract.java

```diff
@@ -24,7 +25,7 @@
  */
 public abstract class ResourceGroupAttributesModuleAbstract extends AttributesModuleAbstract implements ResourceGroupAttributesModuleImplApi{
 
-	public void checkAttributeValue(PerunSessionImpl perunSession, Resource resource, Group group, Attribute attribute) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
+	public void checkAttributeValue(PerunSessionImpl perunSession, Resource resource, Group group, Attribute attribute) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException, GroupResourceMismatchException {
```

Comment:
Please remove ``GroupResourceMismatchException`` and ``MemberResourceMismatchException`` from all member-resource and group-resource attribute modules (api, abstract). If passed entities mismatch, it's programmers fault.
We want these exceptions mostly to be used in outer api (Api and Entry, sometimes it's in Bl/BlImpl).
Then please catch such exceptions in modules implementation and wrap them to ``InternalErrorException``.
This should result in lowering exceptions usages in normal methods from ``AttributesManager`` and back in normal managers.